src/playbacks/html5_video/html5_video.js

```diff
@@ -426,6 +447,96 @@ export default class HTML5Video extends Playback {
     this.trigger(Events.PLAYBACK_READY, this.name)
   }
 
+  _checkForClosedCaptions() {
+    // Check if CC available only if current playback is HTML5Video
+    if (this.isHTML5Video && !this._ccIsSetup) {
+      if (this.hasClosedCaptionsTracks) {
+        this.trigger(Events.PLAYBACK_SUBTITLE_AVAILABLE)
+        const trackId = this.closedCaptionsTrackId
+        this.closedCaptionsTrackId = trackId
+        this.handleTextTrackChange = this._handleTextTrackChange.bind(this)
+        this.el.textTracks.addEventListener('change', this.handleTextTrackChange)
+      }
+      this._ccIsSetup = true
+    }
+  }
+
+  _handleTextTrackChange() {
+    let tracks = this.closedCaptionsTracks
+    let track = tracks.find((track) => {
+      if (track.track.mode === 'showing') {
+        return track
+      }
+    }) || {id: -1}
+    if (this._ccTrackId !== track.id) {
+      this._ccTrackId = track.id
+      this.trigger(Events.PLAYBACK_SUBTITLE_CHANGED, {
+        id: track.id
+      })
+    }
+  }
+
+  get isHTML5Video() {
+    return this.name === HTML5Video.prototype.name
+  }
+
+  get closedCaptionsTracks() {
+    let id = 0
+    let newId = () => { return id++ }
+    let onlySubtitles = (track) => { return track.kind === 'subtitles' }
+    let makeTrack = (track) => { return {id: newId(), name: track.label, track: track} }
+    let textTracks = this.el.textTracks ? Array.from(this.el.textTracks) : []
+
+    return textTracks.filter(onlySubtitles).map(makeTrack)
+  }
+
+  get closedCaptionsTrackId() {
+    return this._ccTrackId
+  }
+
+  set closedCaptionsTrackId(trackId) {
+    if (!isNumber(trackId)) {
+      return
+    }
+
+    let tracks = this.closedCaptionsTracks
+    let showingTrack
+
+    if (trackId !== -1) {
+      // Get track to show
+      showingTrack = tracks.find((track) => {
+        if (track.id == trackId) {
+          return track
+        }
+      })
+    }
+
+    // Ensure track is not already displayed
+    if (showingTrack && showingTrack.track.mode === 'showing') {
+      return
+    }
+
+    // Since it is possible to display multiple tracks,
+    // ensure that all tracks are hidden.
+    tracks
+      .filter((track) => { return track.track.mode !== 'hidden' })
+      .forEach((track) => { track.track.mode = 'hidden' })
+
+    // Display track
+    if (showingTrack) {
+      showingTrack.track.mode = 'showing'
+    } else if (trackId !== -1) {
```

Comment:
-1 means "disabled", ie: if set to -1 if hide subtitles. (_there is a "disabled" label in the select menu_).
nice catch, you are right, i can move up the condition to avoid filter and not change tracks state.